core/backend/src/test/standalone/IModelWrite.test.ts

```diff
@@ -329,6 +331,39 @@ describe("IModelWriteTest", () => {
     rwIModel.close();
   });
 
+  it.only("should set a fake verifyCode for codeService that throws error for operations that affect code, if failed to open codeService ", async () => {
+    const iModelProps = {
+      iModelName: "codeServiceTest",
+      iTwinId,
+    };
+    const iModelId = await HubMock.createNewIModel(iModelProps);
+    const briefcaseProps = await BriefcaseManager.downloadBriefcase({ accessToken: "codeServiceTest", iTwinId, iModelId });
+    const originalCreateForIModel = CodeService.createForIModel;
+    // can be any errors except 'NoCodeIndex'
+    CodeService.createForIModel = async () => {
+      throw new CodeService.Error("MissingCode", 0x10000 + 1, " ");
+    };
+    const briefcaseDb = await BriefcaseDb.open({ fileName: briefcaseProps.fileName});
+    briefcaseDb.channels.addAllowedChannel(ChannelControl.sharedChannelName);
+    let firstElement = {"id":undefined, "codeValue": "test"};
+    briefcaseDb.withPreparedStatement("SELECT * from Bis.Element LIMIT 1", (stmt: ECSqlStatement) => {
+      if (stmt.step() === DbResult.BE_SQLITE_ROW) {
+        firstElement = stmt.getRow();
+      }
+    });
+    // make change to the briefcaseDb that does not affect code, e.g., save file property
+    // expect no error from verifyCode
+    expect(() => briefcaseDb.saveFileProperty({ name: "codeServiceProp", namespace: "codeService", id: 1, subId: 1 }, "codeService test")).to.not.throw();
+    // make change to the briefcaseDb that affects code that will invoke verifyCode, e.g., update an element with a non-null code
+    // expect error from verifyCode
+    const newProps = { id: firstElement.id, code: {...Code.createEmpty(), value:firstElement.codeValue}, classFullName: undefined, model: undefined };
```

Comment:
can you also add a case that we don't throw an error if the Code is an emptyCode somewhere in this test?

Reply:
Added in [commit eec2512af53b9c5b79a8f7fcb99749b1fc7e541a](https://github.com/iTwin/itwinjs-core/pull/6776/commits/eec2512af53b9c5b79a8f7fcb99749b1fc7e541a)